smoke/IotEdgeQuickstart/details/IotedgedLinux.cs

```diff
@@ -200,159 +222,296 @@ public Task Install()
+                // must wait until a request can be processed.
+                while (true)
                 {
-                    await Process.RunAsync("systemctl", "enable iotedge", cts.Token);
-                    await Task.Delay(TimeSpan.FromSeconds(2), cts.Token);
-                    await Process.RunAsync("systemctl", "restart iotedge", cts.Token);
+                    var processInfo = new System.Diagnostics.ProcessStartInfo
+                    {
+                        FileName = "iotedge",
+                        Arguments = "list",
+                        RedirectStandardOutput = true
+                    };
+                    var request = System.Diagnostics.Process.Start(processInfo);
 
-                    // Wait for service to become active
-                    while (true)
+                    if (request.WaitForExit(1000))
                     {
-                        await Task.Delay(TimeSpan.FromSeconds(3), cts.Token);
-                        string[] result = await Process.RunAsync("bash", "-c \"systemctl --no-pager show iotedge | grep ActiveState=\"");
-                        if (result.First().Split("=").Last() == "active")
+                        if (request.ExitCode == 0)
                         {
+                            request.Close();
+                            Console.WriteLine("aziot-edged ready for requests.");
                             break;
                         }
-
-                        errorMessage = result.First();
                     }
-                }
-                catch (OperationCanceledException e)
-                {
-                    throw new Exception($"Error starting iotedged: {errorMessage ?? e.Message}");
+                    else
+                    {
+                        request.Kill(true);
+                        request.WaitForExit();
+                        request.Close();
+                        Console.WriteLine("aziot-edged not yet ready.");
+                    }
                 }
             }
         }
 
         public async Task Stop()
         {
-            // Raspbian's systemctl doesn't support 'disable --now', so do
-            // 'disable' + 'stop' instead
-            await Process.RunAsync("systemctl", "disable iotedge", 60);
-            await Process.RunAsync("systemctl", "stop iotedge", 60);
+            await Process.RunAsync("systemctl", "stop aziot-edged", 60);
+            await Process.RunAsync("systemctl", "stop aziot-identityd", 60);
+            await Process.RunAsync("systemctl", "stop aziot-tpmd", 60);
+            await Process.RunAsync("systemctl", "stop aziot-certd", 60);
+            await Process.RunAsync("systemctl", "stop aziot-keyd", 60);
         }
 
         public Task Reset() => Task.CompletedTask;
+
+        private static async Task<uint> GetIotedgeUid()
+        {
+            string[] output = await Process.RunAsync("id", "-u iotedge");
```

Comment:
Are all the identity services going to be under user `iotedge`?

Reply:
No, each service has its own user. `aziot-edged` runs under the `iotedge` user, so its uid needs to be known to authorize it with `aziot-identityd`.